src/swipes.py

```diff
@@ -0,0 +1,304 @@
+          data_list.append(pd.DataFrame(data={
+              'date': timestamp.strftime('%m/%d/%Y'),
+              'end_time': end_time,
+              'session_type': session_type,
+              'location': location,
+              'start_time': start_time,
+              'swipes': place['CROWD_COUNT'],
+              'weekday': weekday,
+              BRB_ONLY: True if LOCATION_NAMES[location]['type'] == BRB_ONLY else False,
+              DINING_HALL: True if LOCATION_NAMES[location]['type'] == DINING_HALL else False,
+              TRILLIUM: True if LOCATION_NAMES[location]['type'] == TRILLIUM else False,
+              }, index=[0]))
+        i += 1
+        line_counter += 1
+
+      print('done parsing data.log')
+      if 'old_data' in update_info:
+        data_list.append(update_info['old_data'])
+      df = pd.concat(data_list)
+      df.to_csv(MAIN_CSV, header=True, index=False)
+      data_file = sort_by_timeblock(MAIN_CSV)
+      # add new marker to front of table to be used next time for time tracking
+      df = pd.concat([update_info['new_marker'], df])
+      df.to_csv(MAIN_CSV, header=True, index=False)
+      return data_file
+  except Exception as e:
+    print('Failed at parse_to_csv')
+    print('Data update failed:', e)
+
+def sort_by_timeblock(input_file_path, output_file='timeblock-averages.csv'):
+  """
+  Sorts and runs average swipe/time calculations.
+  Saves to a csv file (updates files if already exists):
+  tb-averages.csv - the timeblock swipe averages
+    session_type: type of class session (regular, winter, summer, finals, etc.)
+    weekday: weekday of swipe event, string (ex: monday, tuesday, etc.)
+    location: eatery of swipe event, string (some locations have different names than dining.now's)
+    start_time: left edge of timeblock, string (hh:mm AM/PM)
+    end_time: right edge of timeblock, string (hh:mm AM/PM)
+    swipes: number of swipes, int
+    counter: number of events within this timeblock (used to calculate average), int
+    average: average number of swipes in this timeblock, float (2 decimals)
+    brb_only: indicates if this location is a brb only eatery (used for calcs), boolean
+    dining_hall: indicates if a location is a dining hall (used for calcs), boolean
+    trillium: indicates if a location is Trillium (used for calcs), boolean
+  """
+  try:
+    df = pd.read_csv(input_file_path)
+    # sum together the swipes of rows with same timeblock/eatery makeup
+    df = df.groupby(ISOLATE_SWIPES).sum().reset_index()
+    # count how many days of the same timeblock/eatery combo we have --> used for counter column
+    df = df.groupby(ISOLATE_DATE).count().reset_index()
+    df = df.rename(index=str, columns={'date': 'counter'})
+    # sum together swipes and counters for rows with the same timeblock and location
+    df = df.groupby(ISOLATE_COUNTER_SWIPES).sum().reset_index()
+    df = calculate_wait_times(df)
+    df = df.sort_values(by=['location', 'weekday'])
```

Comment:
writing to the same variable over and over again is no bueno, can these variable names be more verbose

Reply:
I agree that redoing the same variable names is no bueno, This is just how I saw others doing it in examples.  I'll read for better way